app/services/make_an_offer.rb

```diff
@@ -87,15 +114,23 @@ def validate_course_data
     errors.add(:offered_course, "does not belong to provider #{current_provider.code}, it belongs to #{provider.code}") if providers_dont_match
   end
 
-  def validate_offer_conditions
-    return if @offer_conditions.blank?
+  def validate_further_conditions
+    return if further_conditions.blank?
 
-    unless @offer_conditions.is_a?(Array)
-      errors.add(:offer_conditions, 'must be an array')
-      return
+    further_conditions.each_with_index do |value, index|
+      errors.add("further_conditions#{index}", "must be #{MAX_CONDITION_LENGTH} characters or fewer") if value && value.length > MAX_CONDITION_LENGTH
```

Comment:
Does this render out as, for example:

"First condition must be 255 charactes or fewer"

?

Reply:
the inline error message does read that way yes - I've added a screengrab gif

Reply:
The error at the top is not as good as it could be, esp when more than one field is over

![Screenshot 2020-01-02 at 11 54 27](https://user-images.githubusercontent.com/642279/71665936-a52f4f80-2d56-11ea-8acb-2764de129731.png)

Reply:
The red text in the top and next to the field should read:

"First condition must be 255 characters or fewer".

The user shouldn't have to rely on the label text above.

Reply:
@duncanjbrown & @adamsilver I've included the attribute labels in the error messages now:

![image](https://user-images.githubusercontent.com/450843/71673529-f2201f80-2d70-11ea-8a07-3c696163e0a8.png)

![image](https://user-images.githubusercontent.com/450843/71673580-111eb180-2d71-11ea-9ce9-268cde83c8fb.png)